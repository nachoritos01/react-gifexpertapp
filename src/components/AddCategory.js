import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setImputValue] = useState('');
    
    const handleImputChange = (e)=>{
        setImputValue(e.target.value);
        // console.log('handle input change llamado ');

    }

    const handleSubmit = (e)=>{
        e.preventDefault();
        // console.log('hndlessubmit', inputValue);

        if(inputValue.trim().length > 2){

            setCategories(cats=>[ inputValue, ...cats ]);
            setImputValue('');
        }

    }

    return (
        <form onSubmit = {handleSubmit}>
            <p>{ inputValue}</p>

            <input 
                type="text"
                value = {inputValue}
                onChange = {handleImputChange}>

            </input>
        </form>
            
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
