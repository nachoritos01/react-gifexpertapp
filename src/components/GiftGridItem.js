import React from 'react';
import PropTypes from 'prop-types';

export const GiftGridItem = ({title, url}) => {

    return (
        <div className="card animate__animated animate__fadeIn">
            <img alt={title} src={url}/>
            <p>{title}</p>
        </div>
    )
}


GiftGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
