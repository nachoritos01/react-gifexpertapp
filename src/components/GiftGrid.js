import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GiftGridItem } from './GiftGridItem';

export const GiftGrid = ({category}) => {

    //Desustructuracion de los datos de retorno, y renombramos el valos de data
    const {data:images, loading} = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            { loading && <p className="animate__animated animate__fadeIn">Loading</p>}
            <div className="card-grid">
                {/* Mandamos los valores al componente individualmente cap 75 */}
                { images.map((img)=>( <GiftGridItem key={img.id} {... img}/>))}
            </div>
        </>
    )
};

GiftGrid.propTypes = {
    category: PropTypes.string.isRequired
}