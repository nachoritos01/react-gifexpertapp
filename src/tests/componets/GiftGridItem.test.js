import {React} from "react";
import { shallow } from "enzyme";
import { GiftGridItem } from "../../components/GiftGridItem";

describe('Pruebas en <GiftGridItem />', () => {

    //Valores para validar
    const title = 'Un titulo';
    const url = 'https://via.placeholder.com/350x150';

    //Sujeto de pruebas
    const wrapper =shallow(<GiftGridItem title={title} url={url}/>);
    /**
     * Hacer match con el snapshot
     */
    test('Debe de mostratr el componente correctamente ', () => {
        //Esperar que wrapper sea exactamente lo que se esperaba
        expect(wrapper).toMatchSnapshot();
        
    });

    test('Debe de tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title); 
        
    });

    test('Debe de tener la imagen igual al url y el alt ade los props ', () => {

        const img = wrapper.find('img');
        expect( img.prop('src')).toBe(url);
        expect( img.prop('alt')).toBe(title);
        
    });

    test('Debe de tener animate_fadeIn ', () => {
        const div = wrapper.find('div');

        const className = div.prop('className');
        
        const existe = className.search("animate_fadeIn")
        console.log(existe);

        //animate_fadeIn
        expect(existe).toBe(-1);
    });    
    
})
