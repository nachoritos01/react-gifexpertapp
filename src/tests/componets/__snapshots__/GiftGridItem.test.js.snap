// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftGridItem /> Debe de mostratr el componente correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://via.placeholder.com/350x150"
  />
  <p>
    Un titulo
  </p>
</div>
`;
